rest-api/test/unit_test/offline_test/biobank_samples_pipeline_test.py

```diff
@@ -33,11 +40,96 @@ def setUp(self):
     # Everything is stored as a list, so override bucket name as a 1-element list.
     config.override_setting(config.BIOBANK_SAMPLES_BUCKET_NAME, [_FAKE_BUCKET])
     self.participant_dao = ParticipantDao()
+    self.summary_dao = ParticipantSummaryDao()
 
   def _write_cloud_csv(self, file_name, contents_str):
     with cloudstorage_api.open('/%s/%s' % (_FAKE_BUCKET, file_name), mode='w') as cloud_file:
       cloud_file.write(contents_str.encode('utf-8'))
 
+  def _make_biobank_order(self, **kwargs):
+    """Makes a new BiobankOrder (same values every time) with valid/complete defaults.
+
+    Kwargs pass through to BiobankOrder constructor, overriding defaults.
+    """
+    participantId = kwargs['participantId']
+
+    for k, default_value in (
+        ('biobankOrderId', u'1'),
+        ('created', clock.CLOCK.now()),
+        # ('participantId', self.participant.participantId),
+        ('sourceSiteId', 1),
+        ('sourceUsername', u'fred@pmi-ops.org'),
+        ('collectedSiteId', 1),
+        ('collectedUsername', u'joe@pmi-ops.org'),
+        ('processedSiteId', 1),
+        ('processedUsername', u'sue@pmi-ops.org'),
+        ('finalizedSiteId', 2),
+        ('finalizedUsername', u'bob@pmi-ops.org'),
+        ('version', 1),
+        ('identifiers', [BiobankOrderIdentifier(system=u'a', value=u'c')]),
+        ('samples', [BiobankOrderedSample(
+            test=u'1SAL2',
+            description=u'description',
+            processingRequired=True)]),
+        ('dvOrders', [BiobankDVOrder(
+          participantId=participantId
+        )])):
+      if k not in kwargs:
+        kwargs[k] = default_value
```

Comment:
alternative dict merging option for better readability:
```
defaults = {'biobankOrderId': u'1', ...}
kwargs = dict(defaults, **kwargs)
```

Reply:
I copied that code method block from another unittest. Your readability optimization would be a lot better, but I think we should wait until the bit Python 3 re-write to get serious about making code be more pythonic.